src/handler/mod.rs

```diff
@@ -682,11 +693,11 @@ impl Handler {
                 // outgoing session is that we originally sent a RANDOM packet (signifying we did
                 // not have a session for a request) and the packet is not a PING (we are not
                 // trying to update an old session that may have expired.
-                let connection_direction = {
+                let connection_update = {
                     match (request_call.initiating_session(), &request_call.body()) {
-                        (true, RequestBody::Ping { .. }) => ConnectionDirection::Incoming,
-                        (true, _) => ConnectionDirection::Outgoing,
-                        (false, _) => ConnectionDirection::Incoming,
+                        (true, RequestBody::Ping { .. }) => ConnectionUpdate::Incoming,
+                        (true, _) => ConnectionUpdate::Outgoing,
+                        (false, _) => ConnectionUpdate::IncomingIfNotExists,
```

Comment:
Checking this, I agree that the third match is a Incoming only if it doesn't exist, but it seems to be the first one is as well. I think the renaming is worth doing since it better explains what is going on `IncomingIfNotExists` is clearer than incoming in the sense that this is a conditional update.

This means then, removing the `Incoming` variant and adjusting accordingly

Reply:
I have no real opinion on the name change. Happy to do it if we like. 

But I think we probably should also apply the same logic for outgoing connections. If there is already a node that is incoming, we probably don't want to promote it to outgoing. So I guess the options we should do would be:
1. Rename these varients, but probably also `OutgoingIfNotExists`
2. Leave it as it is, and just take the meaning to mean for new connections only. Change the service to only set the connection direction and not modify it (if an ENR already exists)

Reply:
talking with age, the suggestion would be to 
1. remove the conditional that checks the ping (first match statement), so that we depend only on checking if we are initiating the session or not
2. (this one is mine) keep the new names `ConnectionUpdate` definitely is an improvement, and if we want, keep the `*IfNotExists` (or perhaps `*IfNew`) variant names. I find they help with code clarity.
3. As consequence of 1) there would only be two connection updates, the "if new" logic would be entirely moved to the service (as you have already done) but applied to both connection directions
I find this a cleaner solution both in code clarity and cognitive load - goodbye ping special cases!

Can you guys give a thumbs up if we agree?

Reply:
I agree. Just to be clear tho, in the changes we are never updating a connection. We only set the initial connection in the routing table and it now never gets updated. Perhaps the name: `ConnectionUpdate` could be a bit misleading. But it could just be me and I dont mind the name change at all. 
So this sounds good to me.

Reply:
I see your point. I think that us considering this as an update stems from this extra logic that's being moved to the service. In that case, agree on maintaining the original names

Reply:
I agree with the suggestion. I'm also agree with maintaining the original names since in the suggestion we never update connection direction.

Reply:
I have made changes to apply the suggestion and now am doing tests with the simulation locally, please give me a little time.